Yoast/wordpress-seo
# Finish Country Selector Component

## Context


* Adds last required functionality for the SEMrush Country Selector component. 

For the code review it might be nice to look at the commits individually, but please do whatever feels best for you :).

## Summary


This PR can be summarized in the following changelog entry:

* Adds last required functionality for the Country Selector component. 

## Relevant technical choices:

* Adds functionality for the "Change country" button, i.e. when button is used the `semrush_country_code` value in the database is updated
* Renames `database` (redux store) and `currentDatabase` (prop) to `countryCode` to ensure consistent naming across props, redux store and database
* Also renames all `Database` to `Country` as the prop/redux store value/database value are now called `countryCode` and the button states "Change country", to stay consistent and avoid confusion for future developers looking at the code
* Fetches `semrush_country_code` value from database as default for the CountrySelector on page load
* Adds the countryCode to the KeyphrasesTable to include in the "Get more insights link"
* Removes some redundant code

_Unrelated to Country Selector:_
* The `"yoast"` className is moved up in hierarchy to avoid duplicate code, as it was also needed in premium for the buttons ([P3-83](https://yoast.atlassian.net/browse/P3-83)). 
* Since I was already renaming a bunch of things, I also renamed `RelatedKeyphraseModalContent` to the plural `RelatedKeyphrasesModalContent` as it was the only classname with singular form

## Test instructions

This PR can be tested by following these steps:

* Go to this branch and build, also make sure it is linked to the latest version of `feature/semrush` on the monorepo
* Edit a post, the default value for the country selector should be "United States - US"
* Also make sure the redux state `SEMrushRequest` > `countryCode` = "us"
* In the database `wp_options` > `wpseo` > see `semrush_country_code` = "us"
* Select a different country and see the redux action `CHANGE_COUNTRY` and the redux state change 
* Refresh the database and see the `semrush_country_code` value has **not** changed
* Now click the "Change country" button and see the redux action `NEW_REQUEST`
* Refresh the database and see the see the `semrush_country_code` value has changed
* Click the "Get more insights at SEMrush" link and make sure the URL contains: `?q=`*your-keyphrase*`&db=`*your-selected-country*. 

Please make sure everything still works as expected, as quite a few things have been renamed!

Also, right now, when you close the modal and open it again, the country that was last selected is still selected (i.e. not the country from the last request - selected when the "Change Country" button was used). We might need to consider whether this is desired behaviour. When reloading the page, the value from the database is used. 


## UI changes

## Documentation

## Quality assurance


Fixes [P3-82]
